Final list of references in Current  Biology Format
===================
Maria Tocora 
September, 2023

### References in alphabetic order: 
1. Abril, S., and Jurvansuu, J. (2020). Season- and caste-specific variation in RNA viruses in the invasive Argentine ant European supercolony. Journal of General Virology 101, 322–333. 10.1099/jgv.0.001384.
2. Afkhami, M.E., and Stinchcombe, J.R. (2016). Multiple mutualist effects on genomewide expression in the tripartite association between Medicago truncatula, nitrogen‐fixing bacteria and mycorrhizal fungi. Mol Ecol 25, 4946–4962. 10.1111/mec.13809.
3. Altschul, S.F., Gish, W., Miller, W., Myers, E.W., and Lipman, D.J. (1990). Basic local alignment search tool. Journal of Molecular Biology 215, 403–410. 10.1016/S0022-2836(05)80360-2.
4. Amiri, E., Seddon, G., Zuluaga Smith, W., Strand, M.K., Tarpy, D.R., and Rueppell, O. (2019). Israeli Acute Paralysis Virus: Honey Bee Queen–Worker Interaction and Potential Virus Transmission Pathways. Insects 10, 9. 10.3390/insects10010009.
5. Batstone, R.T., Burghardt, L.T., and Heath, K.D. (2022). Phenotypic and genomic signatures of interspecies cooperation and conflict in naturally occurring isolates of a model plant symbiont. Proc. R. Soc. B. 289, 20220477. 10.1098/rspb.2022.0477.
6. Batstone, R.T., O’Brien, A.M., Harrison, T.L., and Frederickson, M.E. (2020). Experimental evolution makes microbes more cooperative with their local host genotype. Science 370, 476–478. 10.1126/science.abb7222.
7. BioBam Bioinformatics (2019). OmicsBox – Bioinformatics Made Easy, BioBam.
8. Blighe, K, Rana, S, and Lewis, M (2018). EnhancedVolcano: Publication-ready volcano plots with enhanced colouring and labeling.
9. Bolger, A.M., Lohse, M., and Usadel, B. (2014). Trimmomatic: a flexible trimmer for Illumina sequence data. Bioinformatics 30, 2114–2120. 10.1093/bioinformatics/btu170.
10. Bushmanova, E., Antipov, D., Lapidus, A., and Prjibelski, A.D. (2019). rnaSPAdes: a de novo transcriptome assembler and its application to RNA-Seq data. GigaScience 8, giz100. 10.1093/gigascience/giz100.
11. Buttstedt, A., Moritz, R.F.A., and Erler, S. (2014). Origin and function of the major royal jelly proteins of the honeybee ( Apis mellifera ) as members of the yellow gene family: Honeybee’s major royals. Biol Rev 89, 255–269. 10.1111/brv.12052.
12. Camacho, C., Coulouris, G., Avagyan, V., Ma, N., Papadopoulos, J., Bealer, K., and Madden, T.L. (2009). BLAST+: architecture and applications. BMC Bioinformatics 10, 421. 10.1186/1471-2105-10-421.
13. Cerenius, L., Kawabata, S., Lee, B.L., Nonaka, M., and Söderhäll, K. (2010). Proteolytic cascades and their involvement in invertebrate immunity. Trends in Biochemical Sciences 35, 575–583. 10.1016/j.tibs.2010.04.006.
14. Kalyaanamoorthy, S., Minh, B.Q., Wong, T.K.F., Von Haeseler, A., and Jermiin, L.S. (2017). ModelFinder: fast model selection for accurate phylogenetic estimates. Nat Methods 14, 587–589. 10.1038/nmeth.4285.
15. Das, B., and De Bekker, C. (2022). Time-course RNASeq of Camponotus floridanus forager and nurse ant brains indicate links between plasticity in the biological clock and behavioral division of labor. BMC Genomics 23, 57. 10.1186/s12864-021-08282-x.
16. de Bekker, C, Will, I, Das, B, and Adams, R.M.M (2018). The ants (Hymenoptera: Formicidae) and their parasites: effects of parasitic manipulations and host responses on ant behavioral ecology. 28, 1–24. https://doi.org/10.25849/myrmecol.news_028:001.
17. Frederickson, M.E. (2013). Rethinking Mutualism Stability: Cheaters and the Evolution of Sanctions. The Quarterly Review of Biology 88, 269–295. 10.1086/673757.
18. Friesen, M.L. (2012). Widespread fitness alignment in the legume–rhizobium symbiosis. New Phytologist 194, 1096–1111. 10.1111/j.1469-8137.2012.04099.x.
19. Fujiyuki, T., Takeuchi, H., Ono, M., Ohka, S., Sasaki, T., Nomoto, A., and Kubo, T. (2004). Novel Insect Picorna-Like Virus Identified in the Brains of Aggressive Worker Honeybees. J Virol 78, 1093–1100. 10.1128/JVI.78.3.1093-1100.2004.
20. Gotz, S., Garcia-Gomez, J.M., Terol, J., Williams, T.D., Nagaraj, S.H., Nueda, M.J., Robles, M., Talon, M., Dopazo, J., and Conesa, A. (2008). High-throughput functional annotation and data mining with the Blast2GO suite. Nucleic Acids Research 36, 3420–3435. 10.1093/nar/gkn176.
21. Gupta, S.K., Kupper, M., Ratzka, C., Feldhaar, H., Vilcinskas, A., Gross, R., Dandekar, T., and Förster, F. (2015). Scrutinizing the immune defence inventory of Camponotus floridanus applying total transcriptome sequencing. BMC Genomics 16, 540. 10.1186/s12864-015-1748-1.
22. Hart, S.F.M., Chen, C.-C., and Shou, W. (2021). Pleiotropic mutations can rapidly evolve to directly benefit self and cooperative partner despite unfavorable conditions. eLife 10, e57838. 10.7554/eLife.57838.
23. Hasegawa, N., Techer, M.A., Adjlane, N., al-Hissnawi, M.S., Antúnez, K., Beaurepaire, A., Christmon, K., Delatte, H., Dukku, U.H., Eliash, N., et al. (2023). Evolutionarily diverse origins of deformed wing viruses in western honey bees. Proc. Natl. Acad. Sci. U.S.A. 120, e2301258120. 10.1073/pnas.2301258120.
24. Hawkings, C., Calkins, T.L., Pietrantonio, P.V., and Tamborindeguy, C. (2019). Caste-based differential transcriptional expression of hexamerins in response to a juvenile hormone analog in the red imported fire ant (Solenopsis invicta). PLoS ONE 14, e0216800. 10.1371/journal.pone.0216800.
25. Henikoff, S., and Henikoff, J.G. (1992). Amino acid substitution matrices from protein blocks. Proc. Natl. Acad. Sci. U.S.A. 89, 10915–10919. 10.1073/pnas.89.22.10915.
26. Hoang, D.T., Chernomor, O., Von Haeseler, A., Minh, B.Q., and Vinh, L.S. (2018). UFBoot2: Improving the Ultrafast Bootstrap Approximation. Molecular Biology and Evolution 35, 518–522. 10.1093/molbev/msx281.
27. Hsu, H.-W., Chiu, M.-C., Lee, C.-C., Lee, C.-Y., and Yang, C.-C.S. (2019). The Association between Virus Prevalence and Intercolonial Aggression Levels in the Yellow Crazy Ant, Anoplolepis Gracilipes (Jerdon). Insects 10, 436. 10.3390/insects10120436.
28. Hsu, H.-W., Chiu, M.-C., Shoemaker, D., and Yang, C.-C.S. (2018). Viral infections in fire ants lead to reduced foraging activity and dietary changes. Sci Rep 8, 13498. 10.1038/s41598-018-31969-3.
29. Huang, Y., Niu, B., Gao, Y., Fu, L., and Li, W. (2010). CD-HIT Suite: a web server for clustering and comparing biological sequences. Bioinformatics 26, 680–682. 10.1093/bioinformatics/btq003.
30. Huerta-Cepas, J., Szklarczyk, D., Heller, D., Hernández-Plaza, A., Forslund, S.K., Cook, H., Mende, D.R., Letunic, I., Rattei, T., Jensen, L.J., et al. (2019). eggNOG 5.0: a hierarchical, functionally and phylogenetically annotated orthology resource based on 5090 organisms and 2502 viruses. Nucleic Acids Research 47, D309–D314. 10.1093/nar/gky1085.
31. Ingram, K.K., Kleeman, L., and Peteru, S. (2011). Differential regulation of the foraging gene associated with task behaviors in harvester ants. BMC Ecol 11, 19. 10.1186/1472-6785-11-19.
32. Ingram, K.K., Oefner, P., and Gordon, D.M. (2005). Task-specific expression of the foraging gene in harvester ants: TASK-SPECIFIC GENE EXPRESSION IN ANTS. Molecular Ecology 14, 813–818. 10.1111/j.1365-294X.2005.02450.x.
33. James W. Baty, Mariana Bulgarella, Jana Dobelmann, Antoine Felden, and Philip J. Lester (2020). Viruses and their effects in ants (Hymenoptera: Formicidae). 10.25849/MYRMECOL.NEWS_030:213.
34. Jones, B.M., Rao, V.D., Gernat, T., Jagla, T., Cash-Ahmed, A.C., Rubin, B.E., Comi, T.J., Bhogale, S., Husain, S.S., Blatti, C., et al. (2020). Individual differences in honey bee behavior enabled by plasticity in brain gene regulatory networks. eLife 9, e62850. 10.7554/eLife.62850.
35. Kanehisa, M. (2000). KEGG: Kyoto Encyclopedia of Genes and Genomes. Nucleic Acids Research 28, 27–30. 10.1093/nar/28.1.27.
36. Kanost, M.R., and Jiang, H. (2015). Clip-domain serine proteases as immune factors in insect hemolymph. Current Opinion in Insect Science 11, 47–55. 10.1016/j.cois.2015.09.003.
37. Katoh, K. (2005). MAFFT version 5: improvement in accuracy of multiple sequence alignment. Nucleic Acids Research 33, 511–518. 10.1093/nar/gki198.
38. Khorramnejad, A., Perdomo, H.D., Palatini, U., Bonizzoni, M., and Gasmi, L. (2021). Cross Talk between Viruses and Insect Cells Cytoskeleton. Viruses 13, 1658. 10.3390/v13081658.
39. Kiers, E.T., Hutton, M.G., and Denison, R.F. (2007). Human selection and the relaxation of legume defences against ineffective rhizobia. Proc. R. Soc. B. 274, 3119–3126. 10.1098/rspb.2007.1187.
40. Koga, R., Moriyama, M., Onodera-Tanifuji, N., Ishii, Y., Takai, H., Mizutani, M., Oguchi, K., Okura, R., Suzuki, S., Gotoh, Y., et al. (2022). Single mutation makes Escherichia coli an insect mutualist. Nat Microbiol 7, 1141–1150. 10.1038/s41564-022-01179-9.
41. Kohlmeier, P., Alleman, A.R., Libbrecht, R., Foitzik, S., and Feldmeyer, B. (2019). Gene expression is more strongly associated with behavioural specialization than with age or fertility in ant workers. Molecular Ecology 28, 658–670. 10.1111/mec.14971.
42. Lester, P.J., Buick, K.H., Baty, J.W., Felden, A., and Haywood, J. (2019). Different bacterial and viral pathogens trigger distinct immune responses in a globally invasive ant. Sci Rep 9, 5780. 10.1038/s41598-019-41843-5.
43. Li, W., and Godzik, A. (2006). Cd-hit: a fast program for clustering and comparing large sets of protein or nucleotide sequences. Bioinformatics 22, 1658–1659. 10.1093/bioinformatics/btl158.
44. Liu, T., Wei, Y., Dong, X., Chen, P., Wang, L., Yang, X., Lu, C., and Pan, M. (2022). The dual roles of three MMPs and TIMP in innate immunity and metamorphosis in the silkworm, Bombyx mori. The FEBS Journal 289, 2828–2846. 10.1111/febs.16313.
45. Love, M.I., Hogenesch, J.B., and Irizarry, R.A. (2016). Modeling of RNA-seq fragment sequence bias reduces systematic errors in transcript abundance estimation. Nat Biotechnol 34, 1287–1291. 10.1038/nbt.3682.
46. Love, M.I., Huber, W., and Anders, S. (2014). Moderated estimation of fold change and dispersion for RNA-seq data with DESeq2. Genome Biol 15, 550. 10.1186/s13059-014-0550-8.
47. Love, M.I., Soneson, C., and Patro, R. (2018). Swimming downstream: statistical analysis of differential transcript usage following Salmon quantification. F1000Res 7, 952. 10.12688/f1000research.15398.3.
48. Lucas, C., and Sokolowski, M.B. (2009). Molecular basis for changes in behavioral state in ant social behaviors. Proc. Natl. Acad. Sci. U.S.A. 106, 6351–6356. 10.1073/pnas.0809463106.
49. Lüthi, M.N., Berardi, A.E., Mandel, T., Freitas, L.B., and Kuhlemeier, C. (2022). Single gene mutation in a plant MYB transcription factor causes a major shift in pollinator preference. Current Biology 32, 5295-5308.e5. 10.1016/j.cub.2022.11.006.
50. Malé, P.-J.G., Turner, K.M., Doha, M., Anreiter, I., Allen, A.M., Sokolowski, M.B., and Frederickson, M.E. (2017). An ant–plant mutualism through the lens of cGMP-dependent kinase genes. Proc. R. Soc. B. 284, 20170896. 10.1098/rspb.2017.0896.
51. Manfredini, F., Riba-Grognuz, O., Wurm, Y., Keller, L., Shoemaker, D., and Grozinger, C.M. (2013). Sociogenomics of Cooperation and Conflict during Colony Founding in the Fire Ant Solenopsis invicta. PLoS Genet 9, e1003633. 10.1371/journal.pgen.1003633.
52. Manni, M., Berkeley, M.R., Seppey, M., and Zdobnov, E.M. (2021). BUSCO: Assessing Genomic Data Quality and Beyond. Current Protocols 1, e323. 10.1002/cpz1.323.
53. Mannino, G., Abdi, G., Maffei, M.E., and Barbero, F. (2018). Origanum vulgare terpenoids modulate Myrmica scabrinodis brain biogenic amines and ant behaviour. PLoS One 13, e0209047. 10.1371/journal.pone.0209047.
54. Mayer, V.E., De Hoog, S., Cristescu, S.M., Vera, L., and Prenafeta-Boldú, F.X. (2021). Volatile Organic Compounds in the Azteca/Cecropia Ant-Plant Symbiosis and the Role of Black Fungi. JoF 7, 836. 10.3390/jof7100836.
55. Mayer, V.E., Frederickson, M.E., McKey, D., and Blatrix, R. (2014). Current issues in the evolutionary ecology of ant–plant symbioses. New Phytologist 202, 749–764. 10.1111/nph.12690.
56. Müller, A.T., Reichelt, M., Cosio, E.G., Salinas, N., Nina, A., Wang, D., Moossen, H., Geilmann, H., Gershenzon, J., Köllner, T.G., et al. (2022). Combined –omics framework reveals how ant symbionts benefit the Neotropical ant-plant Tococa quadrialata at different levels. iScience 25, 105261. 10.1016/j.isci.2022.105261.
57. Negroni, M.A., Segers, F.H.I.D., Vogelweith, F., and Foitzik, S. (2020). Immune challenge reduces gut microbial diversity and triggers fertility-dependent gene expression changes in a social insect. BMC Genomics 21, 816. 10.1186/s12864-020-07191-9.
58. Nguyen, L.-T., Schmidt, H.A., Von Haeseler, A., and Minh, B.Q. (2015). IQ-TREE: A Fast and Effective Stochastic Algorithm for Estimating Maximum-Likelihood Phylogenies. Molecular Biology and Evolution 32, 268–274. 10.1093/molbev/msu300.
59. Nowicka, M., and Robinson, M.D. (2016). DRIMSeq: a Dirichlet-multinomial framework for multivariate count outcomes in genomics. F1000Res 5, 1356. 10.12688/f1000research.8900.2.
60. O’Brien, A.M., Jack, C.N., Friesen, M.L., and Frederickson, M.E. (2021). Whose trait is it anyways? Coevolution of joint phenotypes and genetic architecture in mutualisms. Proc. R. Soc. B. 288, 20202483. 10.1098/rspb.2020.2483.
61. Patro, R., Duggal, G., Love, M.I., Irizarry, R.A., and Kingsford, C. (2017). Salmon provides fast and bias-aware quantification of transcript expression. Nat Methods 14, 417–419. 10.1038/nmeth.4197.
62. Preussger, D., Giri, S., Muhsal, L.K., Oña, L., and Kost, C. (2020). Reciprocal Fitness Feedbacks Promote the Evolution of Mutualistic Cooperation. Current Biology 30, 3580-3590.e7. 10.1016/j.cub.2020.06.100.
63. Pu, Y., Naikatini, A., Pérez‐Escobar, O.A., Silber, M., Renner, S.S., and Chomicki, G. (2021). Genome‐wide transcriptome signatures of ant‐farmed Squamellaria epiphytes reveal key functions in a unique symbiosis. Ecology and Evolution 11, 15882–15895. 10.1002/ece3.8258.
64. Qiu, B., Dai, X., Li, P., Larsen, R.S., Li, R., Price, A.L., Ding, G., Texada, M.J., Zhang, X., Zuo, D., et al. (2022). Canalized gene expression during development mediates caste differentiation in ants. Nat Ecol Evol 6, 1753–1765. 10.1038/s41559-022-01884-y.
65. Qiu, H.-L., Zhao, C.-Y., and He, Y.-R. (2017). On the Molecular Basis of Division of Labor in Solenopsis invicta (Hymenoptera: Formicidae) Workers: RNA-seq Analysis. Journal of Insect Science 17. 10.1093/jisesa/iex002.
66. R Core Team (2017). R: A Language and Environment for Statistical Computing.
67. Razo-Belman, R., Molina-Torres, J., Martínez, O., and Heil, M. (2018). Plant-ants use resistance-related plant odours to assess host quality before colony founding. J Ecol 106, 379–390. 10.1111/1365-2745.12832.
68. Remigi, P., Capela, D., Clerissi, C., Tasse, L., Torchet, R., Bouchez, O., Batut, J., Cruveiller, S., Rocha, E.P.C., and Masson-Boivin, C. (2014). Transient Hypermutagenesis Accelerates the Evolution of Legume Endosymbionts following Horizontal Gene Transfer. PLoS Biol 12, e1001942. 10.1371/journal.pbio.1001942.
69. Riehl, C., and Frederickson, M.E. (2016). Cheating and punishment in cooperative animal societies. Phil. Trans. R. Soc. B 371, 20150090. 10.1098/rstb.2015.0090.
70. Rortais, A., Tentcheva, D., Papachristoforou, A., Gauthier, L., Arnold, G., Colin, M.E., and Bergoin, M. (2006). Deformed wing virus is not related to honey bees’ aggressiveness. Virol J 3, 61. 10.1186/1743-422X-3-61.
71. Roudaire, T., Héloir, M.-C., Wendehenne, D., Zadoroznyj, A., Dubrez, L., and Poinssot, B. (2021). Cross Kingdom Immunity: The Role of Immune Receptors and Downstream Signaling in Animal and Plant Cell Death. Front. Immunol. 11, 612452. 10.3389/fimmu.2020.612452.
72. Rubin, B.E.R., and Moreau, C.S. (2016). Comparative genomics reveals convergent rates of evolution in ant–plant mutualisms. Nat Commun 7, 12679. 10.1038/ncomms12679.
73. Schneider, S.A., Scharffetter, C., Wagner, A.E., Boesch, C., Bruchhaus, I., Rimbach, G., and Roeder, T. (2016). Social stress increases the susceptibility to infection in the ant Harpegnathos saltator. Sci Rep 6, 25800. 10.1038/srep25800.
74. Soneson, C., Love, M.I., and Robinson, M.D. (2015). Differential analyses for RNA-seq: transcript-level estimates improve gene-level inferences. F1000Res 4, 1521. 10.12688/f1000research.7563.1.
75. Srivastava, A., Malik, L., Sarkar, H., Zakeri, M., Almodaresi, F., Soneson, C., Love, M.I., Kingsford, C., and Patro, R. (2020). Alignment and mapping methodology influence transcript abundance estimation. Genome Biol 21, 239. 10.1186/s13059-020-02151-8.
76. Terio, V., Martella, V., Camero, M., Decaro, N., Testini, G., Bonerba, E., Tantillo, G., and Buonavoglia, C. (2008). Detection of a honeybee iflavirus with intermediate characteristics between kakugo virus and deformed wing virus. New Microbiol 31, 439–444.
77. The UniProt Consortium, Bateman, A., Martin, M.-J., Orchard, S., Magrane, M., Agivetova, R., Ahmad, S., Alpi, E., Bowler-Barnett, E.H., Britto, R., et al. (2021). UniProt: the universal protein knowledgebase in 2021. Nucleic Acids Research 49, D480–D489. 10.1093/nar/gkaa1100.
78. Van Den Berge, K., Soneson, C., Robinson, M.D., and Clement, L. (2017). stageR: a general stage-wise method for controlling the gene-level false discovery rate in differential expression and differential transcript usage. Genome Biol 18, 151. 10.1186/s13059-017-1277-0.
79. Vatanparast, M., and Park, Y. (2021). Comparative RNA-Seq Analyses of Solenopsis japonica (Hymenoptera: Formicidae) Reveal Gene in Response to Cold Stress. Genes 12, 1610. 10.3390/genes12101610.
80. Venkataram, S., Kuo, H.-Y., Hom, E.F.Y., and Kryazhimskiy, S. (2023). Mutualism-enhancing mutations dominate early adaptation in a two-species microbial community. Nat Ecol Evol 7, 143–154. 10.1038/s41559-022-01923-8.
81. Venkataraman, S., Prasad, B., and Selvarajan, R. (2018). RNA Dependent RNA Polymerases: Insights from Structure, Function and Evolution. Viruses 10, 76. 10.3390/v10020076.
82. Vilcinskas, A. (2022). Matrix metalloproteinases and their inhibitors – pleiotropic functions in insect immunity and metamorphosis. The FEBS Journal 289, 2805–2808. 10.1111/febs.16314.
83. Von Wyschetzki, K., Lowack, H., and Heinze, J. (2016). Transcriptomic response to injury sheds light on the physiological costs of reproduction in ant queens. Mol Ecol 25, 1972–1985. 10.1111/mec.13588.
84. Weber, M.G., and Agrawal, A.A. (2014). Defense mutualisms enhance plant diversification. Proc. Natl. Acad. Sci. U.S.A. 111, 16442–16447. 10.1073/pnas.1413253111.
85. Weber, M.G., and Keeler, K.H. (2013). The phylogenetic distribution of extrafloral nectaries in plants. Annals of Botany 111, 1251–1261. 10.1093/aob/mcs225.
86. Xavier, C.A.D., Allen, M.L., and Whitfield, A.E. (2021). Ever-increasing viral diversity associated with the red imported fire ant Solenopsis invicta (Formicidae: Hymenoptera). Virol J 18, 5. 10.1186/s12985-020-01469-w.
87. Yek, S.H., Boomsma, J.J., and Schiøtt, M. (2013). Differential gene expression in Acromyrmex leaf-cutting ants after challenges with two fungal pathogen s. Mol Ecol 22, 2173–2187. 10.1111/mec.12255.

