References
===================
Maria Tocora 

1.	A. Alleman, M. Stoldt, B. Feldmeyer, S. Foitzik, Tandem‐running and scouting behaviour are characterized by up‐regulation of learning and memory formation genes within the ant brain. Molecular Ecology 28, 2342–2359 (2019).
2.	S. F. Altschul, W. Gish, W. Miller, E. W. Myers, D. J. Lipman, Basic local alignment search tool. Journal of Molecular Biology 215, 403–410 (1990).
3.	E. Amiri, et al., Israeli Acute Paralysis Virus: Honey Bee Queen–Worker Interaction and Potential Virus Transmission Pathways. Insects 10, 9 (2019).
4.	R. T. Batstone, L. T. Burghardt, K. D. Heath, Phenotypic and genomic signatures of interspecies cooperation and conflict in naturally occurring isolates of a model plant symbiont. Proc. R. Soc. B. 289, 20220477 (2022).
5.	R. T. Batstone, A. M. O’Brien, T. L. Harrison, M. E. Frederickson, Experimental evolution makes microbes more cooperative with their local host genotype. Science 370, 476–478 (2020).
6.	BioBam Bioinformatics, OmicsBox – Bioinformatics Made Easy, BioBam (2019).
7.	Blighe, K, Rana, S, Lewis, M, EnhancedVolcano: Publication-ready volcano plots with enhanced colouring and labeling (2018).
8.	A. M. Bolger, M. Lohse, B. Usadel, Trimmomatic: a flexible trimmer for Illumina sequence data. Bioinformatics 30, 2114–2120 (2014).
9.	Bronstein, JL, Mutualism. (Oxford University Press, New York., 2015).
10.	E. Bushmanova, D. Antipov, A. Lapidus, A. D. Prjibelski, rnaSPAdes: a de novo transcriptome assembler and its application to RNA-Seq data. GigaScience 8, giz100 (2019).
11.	A. Buttstedt, R. F. A. Moritz, S. Erler, Origin and function of the major royal jelly proteins of the honeybee ( Apis mellifera ) as members of the yellow gene family: Honeybee’s major royals. Biol Rev 89, 255–269 (2014).
12.	C. Camacho, et al., BLAST+: architecture and applications. BMC Bioinformatics 10, 421 (2009).
13.	L. Cerenius, S. Kawabata, B. L. Lee, M. Nonaka, K. Söderhäll, Proteolytic cascades and their involvement in invertebrate immunity. Trends in Biochemical Sciences 35, 575–583 (2010).
14.	B. Das, C. De Bekker, Time-course RNASeq of Camponotus floridanus forager and nurse ant brains indicate links between plasticity in the biological clock and behavioral division of labor. BMC Genomics 23, 57 (2022).
15.	C. De Bekker, Ophiocordyceps–ant interactions as an integrative model to understand the molecular basis of parasitic behavioral manipulation. Current Opinion in Insect Science 33, 19–24 (2019).
16.	de Bekker, C, Will, I, Das, B, Adams, R.M.M, The ants (Hymenoptera: Formicidae) and their parasites: effects of parasitic manipulations and host responses on ant behavioral ecology. 28, 1–24 (2018).
17.	B. Epstein, et al., Combining GWAS and population genomic analyses to characterize coevolution in a legume‐rhizobia symbiosis. Molecular Ecology 32, 3798–3811 (2023).
18.	M. E. Frederickson, Rethinking Mutualism Stability: Cheaters and the Evolution of Sanctions. The Quarterly Review of Biology 88, 269–295 (2013).
19.	M. E. Frederickson, et al., The Direct and Ecological Costs of an Ant-Plant Symbiosis. The American Naturalist 179, 768–778 (2012).
20.	M. L. Friesen, Widespread fitness alignment in the legume–rhizobium symbiosis. New Phytologist 194, 1096–1111 (2012).
21.	T. Fujiyuki, et al., Novel Insect Picorna-Like Virus Identified in the Brains of Aggressive Worker Honeybees. J Virol 78, 1093–1100 (2004).
22.	S. D. Gillespie, L. S. Adler, Indirect effects on mutualisms: parasitism of bumble bees and pollination service to plants. Ecology 94, 454–464 (2013).
23.	S. Gotz, et al., High-throughput functional annotation and data mining with the Blast2GO suite. Nucleic Acids Research 36, 3420–3435 (2008).
24.	S. K. Gupta, et al., Scrutinizing the immune defence inventory of Camponotus floridanus applying total transcriptome sequencing. BMC Genomics 16, 540 (2015).
25.	N. Hasegawa, et al., Evolutionarily diverse origins of deformed wing viruses in western honey bees. Proc. Natl. Acad. Sci. U.S.A. 120, e2301258120 (2023).
26.	C. Hawkings, T. L. Calkins, P. V. Pietrantonio, C. Tamborindeguy, Caste-based differential transcriptional expression of hexamerins in response to a juvenile hormone analog in the red imported fire ant (Solenopsis invicta). PLoS ONE 14, e0216800 (2019).
27.	S. Henikoff, J. G. Henikoff, Amino acid substitution matrices from protein blocks. Proc. Natl. Acad. Sci. U.S.A. 89, 10915–10919 (1992).
28.	D. T. Hoang, O. Chernomor, A. Von Haeseler, B. Q. Minh, L. S. Vinh, UFBoot2: Improving the Ultrafast Bootstrap Approximation. Molecular Biology and Evolution 35, 518–522 (2018).
29.	H.-W. Hsu, M.-C. Chiu, C.-C. Lee, C.-Y. Lee, C.-C. S. Yang, The Association between Virus Prevalence and Intercolonial Aggression Levels in the Yellow Crazy Ant, Anoplolepis Gracilipes (Jerdon). Insects 10, 436 (2019).
30.	H.-W. Hsu, M.-C. Chiu, D. Shoemaker, C.-C. S. Yang, Viral infections in fire ants lead to reduced foraging activity and dietary changes. Sci Rep 8, 13498 (2018).
31.	Y. Huang, B. Niu, Y. Gao, L. Fu, W. Li, CD-HIT Suite: a web server for clustering and comparing biological sequences. Bioinformatics 26, 680–682 (2010).
32.	J. Huerta-Cepas, et al., eggNOG 5.0: a hierarchical, functionally and phylogenetically annotated orthology resource based on 5090 organisms and 2502 viruses. Nucleic Acids Research 47, D309–D314 (2019).
33.	K. K. Ingram, L. Kleeman, S. Peteru, Differential regulation of the foraging gene associated with task behaviors in harvester ants. BMC Ecol 11, 19 (2011).
34.	James W. Baty, Mariana Bulgarella, Jana Dobelmann, Antoine Felden, Philip J. Lester, Viruses and their effects in ants (Hymenoptera: Formicidae) (2020) https:/doi.org/10.25849/MYRMECOL.NEWS_030:213 (February 12, 2024).
35.	B. M. Jones, et al., Individual differences in honey bee behavior enabled by plasticity in brain gene regulatory networks. eLife 9, e62850 (2020).
36.	P. Jones, et al., InterProScan 5: genome-scale protein function classification. Bioinformatics 30, 1236–1240 (2014).
37.	S. Kalyaanamoorthy, B. Q. Minh, T. K. F. Wong, A. Von Haeseler, L. S. Jermiin, ModelFinder: fast model selection for accurate phylogenetic estimates. Nat Methods 14, 587–589 (2017).
38.	M. Kanehisa, KEGG: Kyoto Encyclopedia of Genes and Genomes. Nucleic Acids Research 28, 27–30 (2000).
39.	M. R. Kanost, H. Jiang, Clip-domain serine proteases as immune factors in insect hemolymph. Current Opinion in Insect Science 11, 47–55 (2015).
40.	K. Katoh, MAFFT version 5: improvement in accuracy of multiple sequence alignment. Nucleic Acids Research 33, 511–518 (2005).
41.	E. T. Kiers, M. G. Hutton, R. F. Denison, Human selection and the relaxation of legume defences against ineffective rhizobia. Proc. R. Soc. B. 274, 3119–3126 (2007).
42.	R. Koga, et al., Single mutation makes Escherichia coli an insect mutualist. Nat Microbiol 7, 1141–1150 (2022).
43.	P. Kohlmeier, A. R. Alleman, R. Libbrecht, S. Foitzik, B. Feldmeyer, Gene expression is more strongly associated with behavioural specialization than with age or fertility in ant workers. Molecular Ecology 28, 658–670 (2019).
44.	P. J. Lester, K. H. Buick, J. W. Baty, A. Felden, J. Haywood, Different bacterial and viral pathogens trigger distinct immune responses in a globally invasive ant. Sci Rep 9, 5780 (2019).
45.	W. Li, A. Godzik, Cd-hit: a fast program for clustering and comparing large sets of protein or nucleotide sequences. Bioinformatics 22, 1658–1659 (2006).
46.	T. Liu, et al., The dual roles of three MMPs and TIMP in innate immunity and metamorphosis in the silkworm, Bombyx mori. The FEBS Journal 289, 2828–2846 (2022).
47.	M. I. Love, W. Huber, S. Anders, Moderated estimation of fold change and dispersion for RNA-seq data with DESeq2. Genome Biol 15, 550 (2014).
48.	M. I. Love, C. Soneson, R. Patro, Swimming downstream: statistical analysis of differential transcript usage following Salmon quantification. F1000Res 7, 952 (2018).
49.	C. Lucas, M. B. Sokolowski, Molecular basis for changes in behavioral state in ant social behaviors. Proc. Natl. Acad. Sci. U.S.A. 106, 6351–6356 (2009).
50.	M. N. Lüthi, A. E. Berardi, T. Mandel, L. B. Freitas, C. Kuhlemeier, Single gene mutation in a plant MYB transcription factor causes a major shift in pollinator preference. Current Biology 32, 5295-5308.e5 (2022).
51.	P.-J. G. Malé, et al., An ant–plant mutualism through the lens of cGMP-dependent kinase genes. Proc. R. Soc. B. 284, 20170896 (2017).
52.	F. Manfredini, et al., Sociogenomics of Cooperation and Conflict during Colony Founding in the Fire Ant Solenopsis invicta. PLoS Genet 9, e1003633 (2013).
53.	M. Manni, M. R. Berkeley, M. Seppey, E. M. Zdobnov, BUSCO: Assessing Genomic Data Quality and Beyond. Current Protocols 1, e323 (2021).
54.	V. E. Mayer, M. E. Frederickson, D. McKey, R. Blatrix, Current issues in the evolutionary ecology of ant–plant symbioses. New Phytologist 202, 749–764 (2014).
55.	A. T. Müller, et al., Combined –omics framework reveals how ant symbionts benefit the Neotropical ant-plant Tococa quadrialata at different levels. iScience 25, 105261 (2022).
56.	M. A. Negroni, F. H. I. D. Segers, F. Vogelweith, S. Foitzik, Immune challenge reduces gut microbial diversity and triggers fertility-dependent gene expression changes in a social insect. BMC Genomics 21, 816 (2020).
57.	L.-T. Nguyen, H. A. Schmidt, A. Von Haeseler, B. Q. Minh, IQ-TREE: A Fast and Effective Stochastic Algorithm for Estimating Maximum-Likelihood Phylogenies. Molecular Biology and Evolution 32, 268–274 (2015).
58.	M. Nowicka, M. D. Robinson, DRIMSeq: a Dirichlet-multinomial framework for multivariate count outcomes in genomics. F1000Res 5, 1356 (2016).
59.	R. Patro, G. Duggal, M. I. Love, R. A. Irizarry, C. Kingsford, Salmon provides fast and bias-aware quantification of transcript expression. Nat Methods 14, 417–419 (2017).
60.	B. Qiu, et al., Canalized gene expression during development mediates caste differentiation in ants. Nat Ecol Evol 6, 1753–1765 (2022).
61.	R Core Team, R: A Language and Environment for Statistical Computing (2017).
62.	C. Riehl, M. E. Frederickson, Cheating and punishment in cooperative animal societies. Phil. Trans. R. Soc. B 371, 20150090 (2016).
63.	A. Rortais, et al., Deformed wing virus is not related to honey bees’ aggressiveness. Virol J 3, 61 (2006).
64.	B. E. R. Rubin, C. S. Moreau, Comparative genomics reveals convergent rates of evolution in ant–plant mutualisms. Nat Commun 7, 12679 (2016).
65.	J. Sachs, E. Simms, Pathways to mutualism breakdown. Trends in Ecology & Evolution 21, 585–592 (2006).
66.	C. Soneson, M. I. Love, M. D. Robinson, Differential analyses for RNA-seq: transcript-level estimates improve gene-level inferences. F1000Res 4, 1521 (2015).
67.	A. Srivastava, et al., Alignment and mapping methodology influence transcript abundance estimation. Genome Biol 21, 239 (2020).
68.	V. Terio, et al., Detection of a honeybee iflavirus with intermediate characteristics between kakugo virus and deformed wing virus. New Microbiol 31, 439–444 (2008).
69.	The UniProt Consortium, et al., UniProt: the universal protein knowledgebase in 2021. Nucleic Acids Research 49, D480–D489 (2021).
70.	K. Van Den Berge, C. Soneson, M. D. Robinson, L. Clement, stageR: a general stage-wise method for controlling the gene-level false discovery rate in differential expression and differential transcript usage. Genome Biol 18, 151 (2017).
71.	M. Vatanparast, Y. Park, Comparative RNA-Seq Analyses of Solenopsis japonica (Hymenoptera: Formicidae) Reveal Gene in Response to Cold Stress. Genes 12, 1610 (2021).
72.	S. Venkataraman, B. Prasad, R. Selvarajan, RNA Dependent RNA Polymerases: Insights from Structure, Function and Evolution. Viruses 10, 76 (2018).
73.	A. Vilcinskas, Matrix metalloproteinases and their inhibitors – pleiotropic functions in insect immunity and metamorphosis. The FEBS Journal 289, 2805–2808 (2022).
74.	K. Von Wyschetzki, H. Lowack, J. Heinze, Transcriptomic response to injury sheds light on the physiological costs of reproduction in ant queens. Mol Ecol 25, 1972–1985 (2016).
75.	M. G. Weber, A. A. Agrawal, Defense mutualisms enhance plant diversification. Proc. Natl. Acad. Sci. U.S.A. 111, 16442–16447 (2014).
76.	M. G. Weber, K. H. Keeler, The phylogenetic distribution of extrafloral nectaries in plants. Annals of Botany 111, 1251–1261 (2013).
77.	C. A. D. Xavier, M. L. Allen, A. E. Whitfield, Ever-increasing viral diversity associated with the red imported fire ant Solenopsis invicta (Formicidae: Hymenoptera). Virol J 18, 5 (2021).
78.	S. H. Yek, J. J. Boomsma, M. Schiøtt, Differential gene expression in Acromyrmex leaf-cutting ants after challenges with two fungal pathogen s. Mol Ecol 22, 2173–2187 (2013)
