Virus characterization and phylogenetic analysis
==============================
Maria Tocora

We identified genes with GO terms corresponding to viral-related processes (Supplementary Table 3). The genes were manually annotated by aligning them against the NCBI non-redundant protein sequence database using blastx. We found eleven sequences of eight genes from picornavirales and Naranavirus that were homologous to the viral genes we identified. We further extracted potential open reading frames (ORF) and conserved domains by using those genes as queries on NCBI’s ORF Finder and Conserved Domain Database (CDD), respectively. 
For phylogenetic reconstruction, we used the full-length amino acid sequences of the non-structural RNA-dependent RNA polymerase (RdRp), a highly conserved gene suitable for phylogenetic reconstructions of viruses (Venkataraman et al., 2018). We extracted RdRp sequences from genes identified as viral from CDD. Unclassified and classified representative amino acid sequences of species of each family were used in the phylogeny as suggested by Xavier et al. (2021). We also included viral sequences with the most similar matches based on the blastx search. We used noumavirus and three species in the family Secoviridae as outgroups for the phylogenetic placement of picorna-like viruses and narna-like viruses, respectively. We aligned viral protein sequences using MAFFT v7.505 (Katoh, et al., 2005). We generated a maximum likelihood phylogenetic tree based on amino acid sequences with the software program IQ-Tree v1.6.12 software (Nguyen et al., 2015), and the amino acid substitution model Blosum62+R5 (Henikoff and Henikoff, 1992). Finally, we evaluated the support for each node with 1000 bootstrap replicates. 


1. S. Henikoff, J. G. Henikoff, Amino acid substitution matrices from protein blocks. Proc. Natl. Acad. Sci. U.S.A. 89, 10915–10919 (1992).
2. K. Katoh, MAFFT version 5: improvement in accuracy of multiple sequence alignment. Nucleic Acids Research 33, 511–518 (2005).
3. L.-T. Nguyen, H. A. Schmidt, A. Von Haeseler, B. Q. Minh, IQ-TREE: A Fast and Effective Stochastic Algorithm for Estimating Maximum-Likelihood Phylogenies. Molecular Biology and Evolution 32, 268–274 (2015).
4. S. Venkataraman, B. Prasad, R. Selvarajan, RNA Dependent RNA Polymerases: Insights from Structure, Function and Evolution. Viruses 10, 76 (2018).
5. C. A. D. Xavier, M. L. Allen, A. E. Whitfield, Ever-increasing viral diversity associated with the red imported fire ant Solenopsis invicta (Formicidae: Hymenoptera). Virol J 18, 5 (2021).






